lib/galaxy/datatypes/isa.py

```diff
@@ -0,0 +1,520 @@
+        file_start = stream.read(_MAX_LEN_FILE_TYPE_PREFIX)
+        stream.seek(0)  # reset the stream
+        matched_prefix = _FILE_TYPE_REGEX.match(file_start)
+        if matched_prefix:
+            file_type = _FILE_TYPE_PREFIX[matched_prefix.string[matched_prefix.start():matched_prefix.end()]]
+
+        return file_type
+
+    # Set peek {{{2
+    ################################################################
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        """Set the peek and blurb text. Get first lines of the main file and set it as the peek."""
+
+        main_file = self._get_main_file(dataset)
+
+        if main_file is None:
+            raise RuntimeError("Unable to find the main file within the 'files_path' folder")
+
+        # Read first lines of main file
+        with open(main_file, "r") as f:
+            data = []
+            for line in f:
+                if len(data) < _MAX_LINES_HISTORY_PEEK:
+                    data.append(line)
+                else:
+                    break
+            if not dataset.dataset.purged and data:
+                dataset.peek = json.dumps({"data": data})
+                dataset.blurb = 'data'
+            else:
+                dataset.peek = 'file does not exist'
+                dataset.blurb = 'file purged from disk'
+
+    # Display peek {{{2
+    ################################################################
+
+    def display_peek(self, dataset):
+        """Create the HTML table used for displaying peek, from the peek text found by set_peek() method."""
+
+        out = ['<table cellspacing="0" cellpadding="3">']
+        try:
+            if not dataset.peek:
+                dataset.set_peek()
+            json_data = json.loads(dataset.peek)
+            for line in json_data["data"]:
+                line = line.strip()
+                if not line:
+                    continue
+                out.append('<tr><td>%s</td></tr>' % escape(util.unicodify(line, 'utf-8')))
+            out.append('</table>')
+            out = "".join(out)
+        except Exception as exc:
+            out = "Can't create peek %s" % str(exc)
+        return out
+
+    # Generate primary file {{{2
+    ################################################################
+
+    def generate_primary_file(self, dataset=None):
+        """Generate the primary file. It is an HTML file containing description of the composite dataset
+           as well as a list of the composite files that it contains."""
+
+        if dataset:
+            rval = ['<html><head><title>ISA Dataset </title></head><p/>']
+            if hasattr(dataset, "extra_files_path"):
+                rval.append('<div>ISA Dataset composed of the following files:<p/><ul>')
+                for cmp_file in os.listdir(dataset.extra_files_path):
+                    opt_text = ''
+                    rval.append('<li><a href="%s" type="text/plain">%s</a>%s</li>' % (cmp_file, cmp_file, opt_text))
```

Comment:
You escape above but not here, I think should escape these file names and such.

Reply:
I'll escape the displayed filename, not the link, and remove `opt_text`, which is empty.